File: test/Authzi.Tests.MicrosoftOrleans/MicrosoftEntra/MicrosoftEntraID.Common/Initialization/SiloClientBuilder.fs
namespace AuthZI.Tests.MicrosoftOrleans.MicrosoftEntra

open Microsoft.Extensions.DependencyInjection
open Microsoft.Extensions.Hosting
open Orleans.Configuration
open Orleans.Hosting
open System

module SiloClientBuilder =
    let Build (configureDelegate: Action<IServiceCollection>) =
        let hostBuilder = HostBuilder().UseOrleansClient(fun clientBuilder ->
                clientBuilder.UseLocalhostClustering()
                    .Configure<ClusterOptions>(fun (options: ClusterOptions) ->
                        options.ClusterId <- "Orleans.Security.Test"
                        options.ServiceId <- "ServiceId")
                    .ConfigureServices(configureDelegate) |> ignore)
        hostBuilder.Build()

File: test/AuthZI.Tests.MicrosoftOrleans/MicrosoftEntra/MicrosoftEntraID.Common.CSharp/Initialization/SiloClientBuilder.cs
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Orleans.Configuration;
using Orleans.Hosting;
using System;

namespace AuthZI.Tests.MicrosoftOrleans.MicrosoftEntra.MicrosoftEntraID.Common.Initialization;

public static class SiloClientBuilder
{
  public static IHost Build(Action<IServiceCollection> configureDelegate)
  {
    var hostBuilder = new HostBuilder()
      .UseOrleansClient(clientBuilder =>
      {
        clientBuilder
          .UseLocalhostClustering()
          .Configure<ClusterOptions>(options =>
          {
            options.ClusterId = "Orleans.Security.Test";
            options.ServiceId = "ServiceId";
          })
          .ConfigureServices(configureDelegate);
      });

    return hostBuilder.Build();
  }
}